exporter/exporter.go

```diff
@@ -92,57 +105,82 @@ func addHosts(cells []MxCell, scan *core.Scan) []MxCell {
 				As:     "geometry",
 			},
 		})
-		currentY += getHostHeight(&host) + padding
+
+		// Add colored keys
+		keyCount := float32(0)
+		for _, port := range host.Ports {
+			for _, hostKey := range port.HostKeys {
+				if (keyColorMap[hostKey.Fingerprint] != "") {
+					cells = append(cells, makeKeyCell(id, keyColorMap[hostKey.Fingerprint], hostKeyOffsetX, hostKeyOffsetY + (keyHeight + hostKeyPadding) * keyCount))	
+					keyCount += 1
+				}
+			}
+		}
+
+		currentY += getHostHeight(&host) + hostPadding
 		if (i+1)%rows == 0 {
-			currentX += hostWidth + padding
+			currentX += hostWidth + hostPadding
 			currentY = 0
 		}
 	}
 
 	return cells
 }
 
-func addHostsWithSameFingerprint(cells []MxCell, scan *core.Scan) []MxCell {
-	// Group hosts by Fingerprint address
-	hostGroups := make(map[string][]core.Host)
+func addDuplicateHostKeys(cells []MxCell, scan *core.Scan) []MxCell {
```

Comment:
At the moment this function is not only responsible for adding the duplicate host key cells, about also for figuring out what keys are duplicate and storing that info the global key map. This makes the code a bit brittle, as the dev needs to remember now to call `addDuplicateHostKeys` before `addHosts`, so that the global map will be filled correctly. I would instead suggest to create a separate function that only finds out about duplicate host keys and returns the corresponding map. Both the `addHosts` and `addDuplicateHostKeys` can then require this map as a parameter, which should make the dependencies quite obvious.